lib/Doctrine/ODM/MongoDB/Aggregation/Expr.php

```diff
@@ -122,11 +122,10 @@ public function add($expression1, $expression2, ...$expressions): static
      */
     public function addAnd($expression, ...$expressions): static
     {
-        if (! isset($this->expr['$and'])) {
-            $this->expr['$and'] = [];
-        }
-
-        $this->expr['$and'] = array_merge($this->expr['$and'], array_map([$this, 'prepareArgument'], func_get_args()));
+        $this->expr['$and'] = array_merge(
+            $this->expr['$and'] ?? [],
+            array_map([$this, 'prepareArgument'], func_get_args()),
```

Comment:
You could also use a first class callable here and in the other methods where callables are used:
```suggestion
            array_map($this->prepareArgument(...), func_get_args()),
```

Reply:
Good catch, I updated some other places to use named closures.